Objects/listobject.c

```diff
@@ -68,6 +68,20 @@ free_list_items(PyObject** items, bool use_qsbr)
 #endif
 }
 
+static void
+ensure_shared_on_resize(PyListObject *self)
+{
+#ifdef Py_GIL_DISABLED
+    // Ensure that the list array is freed using QSBR if we are not the
```

Comment:
Would it be worth adding an assert to check that the list object is locked?

Reply:
Yeah, that's a good idea. The assert uncovers a some more issues, but I'd like to address those in a follow-up PR so that we can fix the CI sooner.